admin_manual/enterprise/external_storage/s3_swift_as_primary_object_store_configuration.rst

```diff
@@ -1,159 +1,167 @@
-                  // replace key and secret with your credentials
-                  'key' => 'EJ39ITYZEUH5BGWDRUFY',
-                  'secret' => 'M5MrXTRjkyMaxXPe2FRXMTfTfbKEnZCu+7uRTVSj',
-              ],
-          ],
-      ],
-  ],
+The S3 backend mounts a bucket of the Amazon S3 object store into the virtual filesystem.
+The class to be used is ``OCA\Files_Primary_S3\S3Storage``, as in the following example:
+
+.. code-block:: php
+
+   <?php
+
+    $CONFIG = [
+        'objectstore' => [
+            'class' => 'OCA\Files_Primary_S3\S3Storage',
+            'arguments' => [
+                // replace with your bucket
+                'bucket' => 'owncloud',
+                'autocreate' => true,
+                // uncomment to enable server side encryption
+                //'serversideencryption' => 'AES256',
+                'options' => [
+                    // version and region are required
+                    'version' => '2006-03-01',
+                    // change to your region
+                    'region'  => 'eu-central-1',
+                    'credentials' => [
+                        // replace key and secret with your credentials
+                        'key' => 'EJ39ITYZEUH5BGWDRUFY',
+                        'secret' => 'M5MrXTRjkyMaxXPe2FRXMTfTfbKEnZCu+7uRTVSj',
+                    ],
+                ],
+            ],
+       ],
+   ],
 
 
 Ceph S3
 ~~~~~~~
 
-The S3 backend can also be used to mount the bucket of a ceph object store via the s3 API
-into the virtual filesystem. The class to be used is ``OCA\ObjectStore\S3``:
-
-::
-
-    'objectstore' => [
-        'class' => 'OCA\ObjectStore\S3',
-        'arguments' => [
-            // replace with your bucket
-            'bucket' => 'owncloud',
-            'autocreate' => true,
-            'options' => [
-                // version and region are required
-                'version' => '2006-03-01',
-                'region'  => '',
-                // replace key, secret and bucket with your credentials
-                'credentials' => [
-                    // replace key and secret with your credentials
-                    'key'    => 'EJ39ITYZEUH5BGWDRUFY',
-                    'secret' => 'M5MrXTRjkyMaxXPe2FRXMTfTfbKEnZCu+7uRTVSj',
-                ],
-                // replace the ceph endpoint with your rgw url
-                'endpoint' => 'http://cephhost:8000/',
-                // Use path style when talking to ceph
-                'command.params' => [
-                    'PathStyle' => true,
+The S3 backend can also be used to mount the bucket of a Ceph S3 object store via the Amazon S3 API into the virtual filesystem.
+The class to be used is ``OCA\Files_Primary_S3\S3Storage``:
+
+.. code-block:: php
+
+    <?php
+
+    $CONFIG = [
+        'objectstore' => [
+            'class' => 'OCA\Files_Primary_S3\S3Storage',
+            'arguments' => [
+                // replace with your bucket
+                'bucket' => 'OWNCLOUD',
```

Comment:
pls have the bucket name in lower case matchin the other two

Reply:
No. Because ceph and scality use different naming rules

Reply:
good to know, maybe to add a doc hint about that so readers get a hint like:
``The bucket name is in capital letters because of naming rules of ceph and scality.``

Reply:
I'd add a more general hint: read your storage provider/vendor docs on how to create buckets and the naming rules which apply.

Reply:
👍